addon/src/services/date.ts

```diff
@@ -23,3 +23,9 @@ export default class DateService extends Service {
     return Date.parse(dateString);
   }
 }
+
+declare module '@ember/service' {
+  interface Registry {
+    'date': DateService;
+  }
+}
```

Comment:
Why are we manually declaring this vs. using DefinitelyType definitions?

Reply:
Is it because `Registry` uses `extends keyof Registry` for registrations? If so, that makes sense, but I'd prefer those to be in a separate module myself.

Reply:
https://docs.ember-cli-typescript.com/ts/using-ts-effectively#service-and-controller-injections

Reply:
This comment also explains it: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a584575b953b98af9b2d1efd7d65c4719077ddfc/types/ember__service/index.d.ts#L29-L31